.github/workflows/docker-build.yml

```diff
@@ -59,9 +60,9 @@ jobs:
           FLEXFLOW_CONTAINER_TOKEN: ${{ secrets.FLEXFLOW_CONTAINER_TOKEN }}
           FF_GPU_BACKEND: ${{ matrix.gpu_backend }}
         run: |
-          if [[ ${{ github.event_name }} == 'push' && ${GITHUB_REF#refs/heads/} == "master" ]]; then
-            ./docker/publish.sh "flexflow-environment-${FF_GPU_BACKEND}"
-            ./docker/publish.sh "flexflow-${FF_GPU_BACKEND}"
+          if [[ ${{ github.event_name }} == 'push' && (${GITHUB_REF#refs/heads/} == "master" || ${GITHUB_REF#refs/heads/} == "inference") ]]; then
+            # ./docker/publish.sh "flexflow-environment-${FF_GPU_BACKEND}"
+            ./docker/publish.sh "specinfer-${FF_GPU_BACKEND}"
```

Comment:
here, we should make sure that we only name the docker images `specinfer-` when we are in the inference branch, so we should have different arguments to the publish script depending on the name of the branch. I think placing this instruction in the if-statement mentioned above can help with this! 

More specifically, we should have `./docker/publish.sh "specinfer-${FF_GPU_BACKEND}"` for the `inference` branch and `./docker/publish.sh "flexflow-${FF_GPU_BACKEND}"` for the other branches